### PR TITLE
add performance CI

### PR DESCRIPTION
**Pull request by jaimergp**
Trying to get some simple timings with pytest and `--durations`. Might change to `asv` if necessary.

Only triggered with `performance::run` labels

### CODE REVIEW

**Comment by jaimergp** [comment]
Are these timings too good to be true? o.o

osx-64

```
243.27s call     tests/test_performance.py::test_update_python[scipipe.osx-64.lock-classic]
  9.45s call     tests/test_performance.py::test_update_python[scipipe.osx-64.lock-libmamba]
```
win-64

```
116.97s call     tests/test_performance.py::test_update_python[gvleobas.win-64.lock-classic]
  6.49s call     tests/test_performance.py::test_update_python[gvleobas.win-64.lock-libmamba]
```

File: .github/workflows/performance.yml [hunk 1]
```diff
@@ -0,0 +1,185 @@
+name: Performance
+
+on:
+  pull_request:
+    types:
+      - labeled
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (jezdez) [inline .github/workflows/performance.yml:10]:
```suggestion
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number ||
github.sha }}
  cancel-in-progress: true
```

I was only recently pointed out the downside of the default GitHub suggests, that also cancels in-progress tests in the default branch, which may not be something you want to have "tests always pass" in the default branch. Found in https://mail.python.org/archives/list/pypa-committers@python.org/thread/PCBCQMJF64JGRBOX7E2EE4YLKHT4DI55/

File: .github/workflows/performance.yml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+jobs:
+  linux:
+    if: |
+      github.event_name == 'pull_request' &&
+      github.event.label.name == 'performance::run'
```
Comment (jezdez) [inline .github/workflows/performance.yml:16]: Nice!

**Review by jezdez — COMMENTED** [review]
Nice sleuthing!

**Comment by jezdez** [comment]
> Are these timings too good to be true? o.o
> 
> osx-64
> 
> ```
> 243.27s call     tests/test_performance.py::test_update_python[scipipe.osx-64.lock-classic]
>   9.45s call     tests/test_performance.py::test_update_python[scipipe.osx-64.lock-libmamba]
> ```
> 
> win-64
> 
> ```
> 116.97s call     tests/test_performance.py::test_update_python[gvleobas.win-64.lock-classic]
>   6.49s call     tests/test_performance.py::test_update_python[gvleobas.win-64.lock-libmamba]
> ```
> 
> linux-64
> 
> ```
> 352.87s call     tests/test_performance.py::test_update_python[gvleobas.linux-64.lock-classic]
>   4.96s call     tests/test_performance.py::test_update_python[gvleobas.linux-64.lock-libmamba]
> 
> 196.47s call     tests/test_performance.py::test_update_python[scipipe.linux-64.lock-classic]
>   3.34s call     tests/test_performance.py::test_update_python[scipipe.linux-64.lock-libmamba]
> 
>   2.45s call     tests/test_performance.py::test_update_python[mambaforge.linux-64.lock-classic]
>   2.84s call     tests/test_performance.py::test_update_python[mambaforge.linux-64.lock-libmamba]
> ```

Well, we can probably reproduce the slowness?

**Comment by jaimergp** [comment]
Now let's run it again with the verbosity levels down...

**Comment by jaimergp** [comment]
Let's try again, classic choked on Linux 🙊

**Comment by jaimergp** [comment]
Ok, explanation for the Windows differences on `gvleobas` environments:

* `update python`: classic sees it as already solved (as it should, it has the latest Python 3.9 version) and exits early; libmamba solves it again and finds a solution with 3.9.9 (a downgrade from 3.9.10!). I guess `update` is being translated as  `3.9.*` but it should be `>={currently_installed}`? There's some code for this, but we should add a issue and test it thoroughly.
* `install python --update-deps`: classic finds and update for `setuptools` and `tk`; libmamba only for `tk`. `--update-deps` involves a 2nd solver call, and we might have an extra overhead because we are loading _everything_ from scratch again to prevent status leak. classic only calls itself (no new instance). We can improve this but it's a rarely used feature, so low/mid-prio I would say. However it's interesting we don't see this in other OS. Issue should be created.
* `update --all`: could not reproduce locally. They take mostly the same time (19s for libmamba, 23 for classic).

I'll open the mentioned issues and add some more cases for Windows, so this is not our only data point.

**Comment by jaimergp** [comment]
I added one more test, which runs `conda create -c conda-forge -c defaults python=3.9 vaex`. Timings:

linux-64

```
460.66s call     tests/test_performance.py::test_install_vaex_from_conda_forge_and_defaults[classic]
 33.58s call     tests/test_performance.py::test_install_vaex_from_conda_forge_and_defaults[libmamba]
```

osx-64

```
605.89s call     tests/test_performance.py::test_install_vaex_from_conda_forge_and_defaults[classic]
 19.68s call     tests/test_performance.py::test_install_vaex_from_conda_forge_and_defaults[libmamba]
```

win-64

```
501.38s call     tests/test_performance.py::test_install_vaex_from_conda_forge_and_defaults[classic]
 28.53s call     tests/test_performance.py::test_install_vaex_from_conda_forge_and_defaults[libmamba]
```

I feel this better reflects what we usually see :)

**Review by jezdez — APPROVED** [review]
I'm fine with merging this now and iterating on top of it?

**Pull request merged**